lifecyclepolicy/stoppableonce.go

```diff
@@ -0,0 +1,58 @@
+package lifecyclepolicy
+
+import (
+	"sync"
+
+	"github.com/taskcluster/taskcluster-worker/runtime"
+)
+
+// StoppableOnce is a wrapper that ensures we only call StopGracefully and
+// StopNow once and never call StopGracefully after StopNow.
+//
+// There is never any harm in wrapping with this, it merely limits excessive
+// calls to StopNow() and StopGracefully(). Please note that Stoppable.StopNow()
+// may still be invoked after Stoppable.StopGracefully(), it can even be invoked
+// concurrently.
+type StoppableOnce struct {
+	Stoppable          runtime.Stoppable
+	m                  sync.Mutex
+	stoppingNow        chan struct{}
+	stoppingGracefully chan struct{}
+}
+
+// StopGracefully calls StopGracefully() on the s.Stoppable, if neither
+// StopGracefully() or StopNow() have been called.
+func (s *StoppableOnce) StopGracefully() {
```

Comment:
Then how do I get `StopNow` to overwrite `StopGracefully` but not the other way around?